python_modules/dagster/dagster/_utils/caching_instance_queryer.py

```diff
@@ -763,7 +785,12 @@ def get_asset_partitions_updated_after_cursor(
 
         if not updated_after_cursor:
             return set()
-        if after_cursor is None or (not self.asset_graph.is_observable(asset_key)):
+        if after_cursor is None or (
+            # for non-source assets, only examine asset version information if use_asset_versions
```

Comment:
this comment is explaining exactly what the code below does, in the same number of lines, so I don't think it's super useful